src/main/webapp/src/app/ranking/ranking-item/ranking-item.component.ts

```diff
@@ -1,12 +1,30 @@
-import { Component, Input } from '@angular/core';
-import { HiddenGem } from 'src/app/hidden-gem';
+import { Component, Input, OnInit } from '@angular/core';
+import { HiddenGem } from 'src/app/models/hidden-gem';
 
 @Component({
   selector: 'app-ranking-item',
   templateUrl: './ranking-item.component.html',
   styleUrls: ['./ranking-item.component.css']
 })
 
-export class RankingItemComponent {
+export class RankingItemComponent implements OnInit{
   @Input() hiddenGem = {} as HiddenGem;
+  photoUrl!: string;
+  type!: string;
+
+  ngOnInit() {
+    if (this.hiddenGem.photos != null) {
+      this.photoUrl = "https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference="
+        + this.hiddenGem.photos[0].photoReference
+        + "&key=AIzaSyCBb8QQBQal9jDNl3ZG6f3bS6ROX2MtYIM";
+    }
+    else {
+      this.photoUrl = "https://www.flaticon.com/svg/vstatic/svg/3716/3716538.svg?token=exp=1612753174~hmac=f38de8989e705031bdecb2da1464c379"
+    }
+
+    // This prevent the tests from failing with "Cannot read property '0' of undefined"
+    if (this.hiddenGem.types != null) {
```

Comment:
probably better to use !this.hiddenGem.types, that way it catches the type being null or undefined.
(It looks like it might be doing that anyway which surprises me, but I think this way is more obvious)
Also might be worth checking there is at least one item in the types list.

Reply:
I have tried with !this.hiddenGem.types and it still throws the error. So, I revert back to this.hiddenGem.types != null; is that ok?